src/duration.rs

```diff
@@ -35,17 +37,20 @@ impl Duration {
     }
 
     pub fn to_std(&self) -> StdDuration {
-        StdDuration::from_secs_f32(
-            self.year * 60. * 60. * 24. * 30. * 12.
-                + self.month * 60. * 60. * 24. * 30.
-                + self.day * 60. * 60. * 24.
-                + self.hour * 60. * 60.
-                + self.minute * 60.
-                + self.second,
+        let millis = (self.second.fract() * 1000.0).round() as u64;
+        StdDuration::from_millis(
+            (self.year.round() as u64 * YEAR_IN_S
+                + self.month.round() as u64 * 60 * 60 * 24 * 30 // there is no official answer on how long a month is, so 30 days will have to do
+                + self.day.round() as u64 * 60 * 60 * 24
+                + self.hour.round() as u64 * 60 * 60
```

Comment:
This will round 1.5 hours to 2 hours.
How about `(self.hour * 60.0 * 60.0) as u64` ?

Reply:
Right. I forgot that the actual goal was to support fractional times. :)
Fundamentally, I think converting everything to integer seconds then converting back to X is the correct approach.